doc/api/dgram.md

```diff
@@ -239,8 +239,9 @@ respect to [byte length][] and not the character position.
 If `msg` is an array, `offset` and `length` must not be specified.
 
 The `address` argument is a string. If the value of `address` is a host name,
-DNS will be used to resolve the address of the host. If the `address` is not
-specified or is an empty string, `'127.0.0.1'` or `'::1'` will be used instead.
+DNS will be used to resolve the address of the host. Otherwise, if `address` is
+not specified or is an empty string, `'127.0.0.1'` (`udp4`) or `'::1'` (`udp6`)
```

Comment:
I can change the wording to "provide". 

For an `address` that is not provided, we defer to the accepted types in `dns.lookup()`.  Per the code there, `address` is allowed to be a *falsy* value, or a string.

So, I may have been a bit too stringent in my checks.  The allowed values of `address` are then:

1.  a nonempty string (which is considered "provided"; the rest are not)
2. `false`
3. `undefined`
4. `null`
5. `0`
6. `''`

If it's any other truthy value (except `Function`; see below), it's invalid, and `send()` should throw a `TypeError`.

Furthermore, if `address` *is* a `Function`, then it's the `callback`, and the `address` will be reset to a falsy value (I think it was `undefined`, but it doesn't really matter).   

"Falsy" and "truthy" seem to be unique to JS and there's precedence in the Node.js docs for their usage.  Explicitly enumerating all "falsy" values is cumbersome, anyway.

I propose the following changes, then:

1.  Avoid ambiguity by stating "If `address` is a non-empty string, use it to resolve the host.  If `address` is falsy, `127.0.0.1` (etc) will be used instead.`
2.  Update my type check(s) to allow all falsy values to pass through to `dns.lookup()`.
3.  Revisit tests to ensure the "falsy" values *don't* throw exceptions, and truthy non-strings *do* throw exceptions.

Sound fair?